### ISSUE TITLE
:format linter doesn't handle `logf` correctly

### ISSUE THREAD

**Issue by ihumrt** [clj-kondo/clj-kondo#2712]
The `:format` linter correctly handles `errorf`, `infof` and `debugf` (in that it knows that the format string can be the 0th or 1st arg, due to the optional `Throwable` arg). But it also handles `logf` using the exact same logic, even though `logf` takes one additional argument. This means that:

```clojure
(errorf e "id: %s") ;; "Format string expects 1 argument instead of 0"
(logf :error e "id: %s") ;; no warning
```

The second line above should issue the same warning as the first line.

**Comment by borkdude** [comment]
Thanks
